maestro-cli/src/main/java/maestro/cli/api/ApiClient.kt

```diff
@@ -0,0 +1,142 @@
+package maestro.cli.api
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.github.michaelbull.result.Err
+import com.github.michaelbull.result.Ok
+import com.github.michaelbull.result.Result
+import com.github.michaelbull.result.map
+import maestro.cli.CliError
+import okhttp3.MediaType.Companion.toMediaType
+import okhttp3.MultipartBody
+import okhttp3.OkHttpClient
+import okhttp3.Protocol
+import okhttp3.Request
+import okhttp3.RequestBody.Companion.asRequestBody
+import okhttp3.RequestBody.Companion.toRequestBody
+import okhttp3.Response
+import java.nio.file.Path
+import java.util.concurrent.TimeUnit
+import kotlin.io.path.absolutePathString
+import kotlin.io.path.exists
+
+class ApiClient(
+    private val baseUrl: String,
+) {
+
+    private val client = OkHttpClient.Builder()
+        .readTimeout(5, TimeUnit.MINUTES)
+        .writeTimeout(5, TimeUnit.MINUTES)
+        .protocols(listOf(Protocol.HTTP_1_1))
+        .build()
+
+    fun magicLinkLogin(email: String): Result<String, Response> {
+        return post<Map<String, Any>>("/magiclink/login", mapOf(
+            "deviceId" to "",
+            "email" to email,
+        )).map { it["requestToken"].toString() }
+    }
+
+    fun magicLinkSignUp(email: String, teamName: String): Result<String, Response> {
+        return post("/magiclink/signup", mapOf(
+            "deviceId" to "",
+            "userEmail" to email,
+            "teamName" to teamName,
+        ))
+    }
+
+    fun magicLinkGetToken(requestToken: String): Result<String, Response> {
+        return post<Map<String, Any>>("/magiclink/gettoken", mapOf(
+            "requestToken" to requestToken,
+        )).map { it["authToken"].toString() }
+    }
+
+    fun isAuthTokenValid(authToken: String): Boolean {
+        val request = try {
+            Request.Builder()
+                .get()
+                .header("Authorization", "Bearer $authToken")
+                .url("$baseUrl/auth")
+                .build()
+        } catch (e: IllegalArgumentException) {
+            if (e.message?.contains("Unexpected char") == true) {
+                return false
+            } else {
+                throw e
+            }
+        }
+        val response = client.newCall(request).execute()
+        return response.isSuccessful
+    }
+
+    fun upload(
+        authToken: String,
+        appFile: Path,
+        workspaceZip: Path,
+        uploadName: String,
+        repoOwner: String?,
+        repoName: String?,
+        branch: String?,
+        pullRequestId: String?,
+    ): UploadResponse {
+        if (!appFile.exists()) throw CliError(appFile.absolutePathString())
+        if (!workspaceZip.exists()) throw CliError(workspaceZip.absolutePathString())
```

Comment:
These throw a `CliError` with the path, shouldn't they also tell the reason why? For instance, `App file does not exist: ${appFile.absolutePathString()}`?

Reply:
Good catch! These used to be more descriptive exceptions and after converting to a CLIError I forgot to update the message